src/write.rs

```diff
@@ -176,6 +179,13 @@ impl<W> WavWriter<W>
             data_bytes_written: 0,
             sample_writer_buffer: Vec::new(),
             finalized: false,
+            // Tip from https://msdn.microsoft.com/en-us/library/ms713497.aspx:
+            // Formats that support more than two channels or sample sizes of more
+            // than 16 bits can be described in a WAVEFORMATEXTENSIBLE structure,
+            // which includes the WAVEFORMAT structure.
+            extensible: spec.channels > 2 ||
+                        spec.bits_per_sample > 16 ||
+                        spec.sample_format == SampleFormat::Float,
```

Comment:
The `write_waveformatex` method below does handle a float sample format, this check is not required here, is it?